internal/cloud/awscloud/maintenance.go

```diff
@@ -84,6 +85,31 @@ func (a *AWS) DescribeInstancesByTag(tagKey, tagValue string) ([]ec2types.Reserv
 	return res.Reservations, nil
 }
 
+func (a *AWS) DescribeInstancesByTag(tagKey, tagValue string) ([]ec2types.Reservation, error) {
+	return a.describeInstancesByKeyValue(fmt.Sprintf("tag:%s", tagKey), tagValue)
+}
+
+func (a *AWS) DescribeInstancesBySecurityGroupID(securityGroupID string) ([]ec2types.Reservation, error) {
+	return a.describeInstancesByKeyValue("instance.group-id", securityGroupID)
+}
+
+func (a *AWS) DescribeInstancesByLaunchTemplateID(launchTemplateID string) ([]ec2types.Reservation, error) {
+	return a.describeInstancesByKeyValue("tag:aws:ec2launchtemplate:id", launchTemplateID)
```

Comment:
This looks weird - I'll check if there is another option to get the instances (or do you know by heart how it works @croissanne ?)

Reply:
Eh, looks ok to me. https://awscli.amazonaws.com/v2/documentation/api/2.0.33/reference/ec2/describe-instances.html#options are the filters, and i don't see launchtemplate